### DISCUSSION TITLE
How to build termux environment on aarch64 server?

### DISCUSSION THREAD

**Discussion by txmu** [termux/termux-packages#7888] [Q&A] [answered]
I have a high-performance server of aarch64. How can I use it to compile the software available for termux, or do other things about termux?

**Comment by agnostic-apollo** [comment, marked as answer]
https://github.com/termux/termux-packages/wiki/Build-environment

https://github.com/termux/termux-packages/wiki/Building-packages

**Reply by txmu** [reply]
I tried to use Docker or set up the environment in the host, but failed, so I have this problem

**Reply by txmu** [reply]
I tried to use Docker or set up the environment in the host, but failed, so I have this problemp

**Reply by txmu** [reply]
I have tried the steps on these pages a few days ago

**Reply by agnostic-apollo** [reply]
Failed with what error?

The termux package maintainers and contributors use the same instructions for building packages locally before pushing code. There might be some issues due to ndk updates in last few days/weeks. Try pulling latest changes in repo and updating docker image and trying again. If some error occurs, you may open an issue about it with the **exact error** you are getting.
